vowpalwabbit/array_parameters.h

```diff
@@ -0,0 +1,172 @@
+#pragma once
```

Comment:
Hi Aarti,

I have been looking at your pull request, and it's a really nice piece of work.

I have a fair bit of experience writing C++ but am only getting started working with VW.  As such, I would like to offer some suggestions regarding design, coding style, and a few other observations.  They will range from from (hopefully!) useful, to nitpicky.

As such, please feel free to take on as much or as little of what I write.  From now on, I will keep my comments very short, but please don't mistake my brevity for criticism.  

My first comment will be a bit of a nitpick.  #pragma once is not standard C++, and as such, it isn't guaranteed to work on all compilers.  It turns out that it works well on Visual C++, but has historically not worked well on gcc.

It looks like the rest of the project is using it, so you are perhaps right to stick to the style.  At the same time, when writing completely new code, it may be worthwhile evolving the current style.  :D